invenio/legacy/oairepository/server.py

```diff
@@ -62,10 +62,13 @@
 from intbitset import intbitset
 from invenio.utils.html import X, EscapedXMLString
 from invenio.legacy.dbquery import run_sql, wash_table_column_name
-from invenio.legacy.search_engine import record_exists, get_all_restricted_recids, get_all_field_values, search_unit_in_bibxxx, get_record, search_pattern
+from invenio.legacy.search_engine import record_exists, get_all_restricted_recids, \
+    get_all_field_values, search_unit_in_bibxxx, get_record, search_pattern, \
+    get_records_that_can_be_displayed
 from invenio.modules.formatter import format_record
 from invenio.legacy.bibrecord import record_get_field_instances
 from invenio.ext.logging import register_exception
+from flask_login import current_user
```

Comment:
please check with @lnielsen about the `current_user` usage and in case it's fine please re-order the imports.